database/transaction.rs

```diff
@@ -144,11 +144,7 @@ impl<D: DurabilityClient> TransactionWrite<D> {
 
     pub fn try_commit(self) -> Result<(), DataCommitError> {
         let mut snapshot = Arc::into_inner(self.snapshot).ok_or_else(|| DataCommitError::SnapshotInUse {})?;
-        self.thing_manager.finalise(&mut snapshot).map_err(|errs| {
-            // TODO: send all the errors, not just the first
-            let error = errs.into_iter().next().unwrap();
-            DataCommitError::ConceptWriteErrorsFirst { typedb_source: error }
-        })?;
+        self.thing_manager.finalise(&mut snapshot).map_err(|source| DataCommitError::ConceptWriteErrors { source })?;
```

Comment:
@flyingsilverfin Is there anything else needed to do here? Maybe I don't get the idea of this TODO.

Reply:
Basically: we have this stack trace system, where if you provide a `typedb_source`, we can print it in a stack trace with useful details. However, this only works with 1 error, not many (otherwise it will be a stack trace tree) - keep the old error and TODO please!

Reply:
Ah... I'll extend the todo, because it just felt like we need to change the return type...

Reply:
Done